src/items/functions.md

```diff
@@ -160,6 +160,38 @@ attributes], [`must_use`], [the procedural macro attributes], [the testing
 attributes], and [the optimization hint
 attributes].
 
+## Const functions
+
+Functions can be `const`, meaning they can be called from within array length expressions and the initializer of constants, statics and enum discriminants. When called from such a so-called "const context", the function is interpreted by the compiler at compile time. The interpretation happens in the environment of the compilation target and not the host. So `usize` is `32` bits if you are compiling against a `32` bit system, irrelevant of whether you are building on a `64` bit or a `32` bit system.
```

Comment:
Let's actually give a name and definition with this. You informally call it `` `const fn` `` in future paragraphs, but the name should be prose, not code, so I think "const function" or "constant function" (or both?) works. We should also be more explicit about how they are defined. I propose the following, but feel to tweak it, using "constant function" and using the length of this comment box for where to put in newlines instead of 80 chars...

```markdown
*Constant functions* are functions that can be called in [const context]. They
are defined by putting the `const` keyword before the `fn` keyword.
```

Reply:
I want to avoid calling these "constant functions" because it gives the impression that the functions are somehow constant; they aren't. They are *pure* functions (at least currently and besides some weird business with floating points). They also can't always be called in `const ITEM = the_const_fn(...);` contexts if given runtime arguments and they can always be called in runtime contexts.

If you must have a name in prose (which strikes me as less formal) then I would call it "const functions".

Reply:
> They also can't always be called in const ITEM = the_const_fn(...); contexts if given runtime arguments and they can always be called in runtime contexts.

The `constant expression` section currently states "The following expressions are constant expressions, so long as any operands are also constant expressions and do not cause any `Drop::drop` calls to be ran.". It's not the most precise wording, but it's supposed to handle that.

> they can always be called in runtime contexts

All constant expressions can be.

> If you must have a name in prose (which strikes me as less formal)

We want to have the name for other documentation none-the-less. And we want the name so that we can then discuss it in the prose further on.

Reply:
This one isn't addressed.